src/inner_op/gridshift.rs

```diff
@@ -4,74 +4,101 @@ use crate::authoring::*;
+
+        for grid in grids.iter() {
+            if let Some(d) = grid.interpolation(&coord, None) {
+                // Geoid
+                if grid.bands() == 1 {
+                    coord[2] -= d[0];
+                    operands.set_coord(i, &coord);
+                    successes += 1;
+
+                    continue 'points;
+                }
+
+                // Datum shift
+                coord[0] += d[0];
+                coord[1] += d[1];
+                operands.set_coord(i, &coord);
+                successes += 1;
+
+                continue 'points;
+            }
+        }
+
+        if use_null_grid {
             successes += 1;
-            operands.set_coord(i, &coord);
+            continue;
         }
-        return successes;
-    }
 
-    // Datum shift
-    for i in 0..n {
-        let mut coord = operands.get_coord(i);
-        let d = grid.interpolation(&coord, None);
-        coord[0] += d[0];
-        coord[1] += d[1];
-        operands.set_coord(i, &coord);
-        successes += 1;
+        // No grid found so we stomp on the coordinate
+        operands.set_coord(i, &Coor4D::nan());
     }
+
     successes
 }
 
 // ----- I N V E R S E --------------------------------------------------------------
 
 fn inv(op: &Op, _ctx: &dyn Context, operands: &mut dyn CoordinateSet) -> usize {
-    let grid = &op.params.grids["grid"];
+    let grids = &op.params.grids;
+    let use_null_grid = op.params.boolean("null_grid");
+
     let mut successes = 0_usize;
     let n = operands.len();
 
-    // Geoid
-    if grid.bands == 1 {
-        for i in 0..n {
-            let mut coord = operands.get_coord(i);
-            let t = grid.interpolation(&coord, None);
-            coord[2] += t[0];
-            operands.set_coord(i, &coord);
-            successes += 1;
-        }
-        return successes;
-    }
+    'points: for i in 0..n {
+        let mut coord = operands.get_coord(i);
 
-    // Datum shift - here we need to iterate in the inverse case
-    for i in 0..n {
-        let coord = operands.get_coord(i);
-        let mut t = coord - grid.interpolation(&coord, None);
-
-        for _ in 0..10 {
-            let d = t - coord + grid.interpolation(&t, None);
-            t = t - d;
-            // i.e. d.dot(d).sqrt() < 1e-10
-            if d.dot(d) < 1e-20 {
-                break;
+        for grid in grids.iter().rev() {
+            if let Some(t) = grid.interpolation(&coord, None) {
+                // Geoid
+                if grid.bands() == 1 {
+                    coord[2] += t[0];
+                    operands.set_coord(i, &coord);
+                    successes += 1;
+
+                    continue 'points;
+                }
+
+                // Datum shift - here we need to iterate in the inverse case
+                let mut t = coord - t;
+
+                for _ in 0..10 {
+                    // NOTE: Is it safe to unwrap on subsequent interpolations?
```

Comment:
@busstoptaktik to check before we finalise, is this safe to do?

Reply:
No, I think there's some additional cleanup to do here, because the iteration may start (or wander) outside of the grid coverage.

The 100% right solution will require some decisions and quite a number of changes, so I believe that once we wander outside of the grid / all grids, we should (for now, and as future default) just set `t = coord, successes+=1` and continue, if `use_null_grid`, or set `t = nan()` and continue if not.

Reply:
@busstoptaktik could you take another look please. It's not strictly necessary but I've labelled the iterate loop for clarity more than anything given we've got three nested loops here.